src/Server/Services/Scp/ClaraAeChangedNotificationService.cs

```diff
@@ -0,0 +1,100 @@
+/*
+ * Apache License, Version 2.0
+ * Copyright 2019-2021 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Ardalis.GuardClauses;
+using Microsoft.Extensions.Logging;
+using Nvidia.Clara.DicomAdapter.API;
+using Nvidia.Clara.DicomAdapter.Server.Common;
+using System;
+using System.Collections.Generic;
+
+namespace Nvidia.Clara.DicomAdapter.Server.Services.Scp
+{
+    public enum ChangedEventType
+    {
+        Added,
+        Updated,
+        Deleted
+    }
+
+    public class ClaraApplicationChangedEvent
+    {
+        public ClaraApplicationEntity ApplicationEntity { get; }
+
+        public ChangedEventType Event { get; }
+
+        public ClaraApplicationChangedEvent(ClaraApplicationEntity applicationEntity, ChangedEventType eventType)
```

Comment:
validate `applicationEntity` and `eventType`?

Reply:
Fixed.